wolfcrypt/src/port/Renesas/renesas_tsip_aes.c

```diff
@@ -151,6 +168,349 @@ int wc_tsip_AesCbcDecrypt(struct Aes* aes, byte* out, const byte* in, word32 sz)
 }
+/* 
+ * Encrypt plain data then output encrypted data and authentication tag data.
+ * The session key used for encryption is generated inside this function and 
+ * the key which has been generated and stored in Aes is not used.
+ * parameter
+ *  - aes:  Aes structure
+ *  - out:  buffer where the cipher text is output
+ *  - in:   buffer where the plain data is storead
+ *  - sz:   size of plain data and also means output size
+ *  - iv:   iv should be consist of implicit-iv of 4 bytes and exp-iv of 8 bytes
+ *  - authTag: buffer where the authentication data is output.
+ *  - authTagSz: buffer size for authentication data.
+ *  - authIn: buffer holding Additionnal Authentication Data(AAD)
+ *  - authInSz: AAD size
+ *  - ctx:   TsipUserCtx
+ * return 0 on success, otherwise on error. 
+ * Note: As of TSIPv1.13, only accept 128 and 256 bit of key size 
+ * 
+ */
+int wc_tsip_AesGcmEncrypt(
+    struct Aes*  aes,     byte* out,
+    const  byte* in,       word32 sz,
+           byte* iv,       word32 ivSz,
+           byte* authTag,  word32 authTagSz, /* auth Tag */
+    const  byte* authIn,   word32 authInSz,  /* AAD */
+           void* ctx)
+{
+    int                 ret = -1;
+    e_tsip_err_t        err;
+    tsip_gcm_handle_t   hdl;
+    uint32_t            dataLen = sz;
+    uint32_t            cipherBufSz;
+
+    aesGcmEncInitFn     initFn;
+    aesGcmEncUpdateFn   updateFn;
+    aesGcmEncFinalFn    finalFn;
+
+    uint8_t* plainBuf  = NULL;
+    uint8_t* cipherBuf = NULL;
+    uint8_t* aTagBuf   = NULL;
+    uint8_t* aadBuf    = NULL;
+
+    tsip_aes_key_index_t key_client_aes;
+    TsipUserCtx *userCtx;
+
+    WOLFSSL_MSG(">> wc_tsip_AesGcmEncrypt");
+
+    if (aes == NULL || ctx == NULL ||
+       (sz != 0       && (in == NULL || out == NULL)) || 
+       (ivSz != 0     &&  iv == NULL) ||
+       (ivSz < AESGCM_NONCE_SZ && iv != NULL) ||  /* Requires 12 bytes of iv */
+       (authInSz != 0 && authIn == NULL) ||
+       (authTagSz < AES_BLOCK_SIZE)) {
+       WOLFSSL_MSG("wc_tsip_AesGcmEncrypt: illegal args");
+       return BAD_FUNC_ARG;
+    }
+    /* TSIP can handle 128 and 256 bit key only */
+    if (aes->ctx.keySize != 16 && aes->ctx.keySize != 32) {
+        WOLFSSL_MSG("wc_tsip_AesGcmEncrypt: illegal key size");
+        return  BAD_FUNC_ARG;
+    }
+
+    if (aes->ctx.keySize == 16) {
+        initFn   = R_TSIP_Aes128GcmEncryptInit;
+        updateFn = R_TSIP_Aes128GcmEncryptUpdate;
+        finalFn  = R_TSIP_Aes128GcmEncryptFinal;
+    }
+    else {
+        initFn   = R_TSIP_Aes256GcmEncryptInit;
+        updateFn = R_TSIP_Aes256GcmEncryptUpdate;
+        finalFn  = R_TSIP_Aes256GcmEncryptFinal;
+    }
+
+    userCtx = (TsipUserCtx*)ctx;
+
+    /* buffer for cipher data outout must be multiple of AES_BLOCK_SIZE */
+    cipherBufSz = ((sz / AES_BLOCK_SIZE) + 1) * AES_BLOCK_SIZE;
+    
+    if ((ret = tsip_hw_lock()) == 0) {
+
+        /* allocate buffers for plaintaxt, ciphertext, authTag and aad to make
```

Comment:
Spelling - "plaintaxt" should be "plaintext"

Reply:
Please address

Reply:
Fixed it.